package com.capgemini.studentexception.tests;

import static org.junit.Assert.assertNotNull;

import org.junit.Test;

import com.capgemini.studentexception.exception.CountryException;
import com.capgemini.studentexception.exception.NameException;
import com.capgemini.studentexception.main.TaxCalculator;

public class TaxCalculatorTest {

	@Test
	public void testToCheckObjectIsCreatedByDefaultConstructor() {
		TaxCalculator taxCalculator = new TaxCalculator();
		assertNotNull(taxCalculator);
	}

	@Test
	public void testToCheckValidCountry() throws CountryException, NameException {
		TaxCalculator taxCalculator = new TaxCalculator("Lahari", "Indian", 150000);

	}

	@Test(expected = CountryException.class)
	public void testToCheckNotValidCountry() throws CountryException, NameException {
		TaxCalculator taxCalculator = new TaxCalculator("Lahari", "America", 150000);

	}

	@Test
	public void testToCheckValidName() throws CountryException, NameException {
		TaxCalculator taxCalculator = new TaxCalculator("Lahari", "Indian", 150000);
	}
	@Test
	public void testToCheckNotValidName() throws CountryException, NameException {
		TaxCalculator taxCalculator = new TaxCalculator("", "America", 150000);

	}
}
